### PR TITLE
Added ESLint upgrade instructions to middleware upgrade guide

### PR DESCRIPTION
**Pull request by steven-tey**
## Bug

- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Errors have helpful link attached, see `contributing.md`

## Feature

- [ ] Implements an existing feature request or RFC. Make sure the feature request has been accepted for implementation before opening a PR.
- [ ] Related issues linked using `fixes #number`
- [ ] Integration tests added
- [ ] Documentation added
- [ ] Telemetry added. In case of a feature if it's used or not.
- [ ] Errors have helpful link attached, see `contributing.md`

## Documentation / Examples

- [ ] Make sure the linting passes by running `pnpm lint`
- [ ] The examples guidelines are followed from [our contributing doc](https://github.com/vercel/next.js/blob/canary/contributing.md#adding-examples)

### CODE REVIEW

File: errors/middleware-upgrade-guide.md [hunk 1]
```diff
@@ -9,6 +9,8 @@ This upgrade guide will help you understand the changes and how to migrate your
 
 You can start upgrading your Middleware usage today with the latest canary release (`npm i next@canary`).
 
+If you have ESLint configured, you will need to run `npm i eslint-config-next@canary --force` to upgrade your ESLint configuration to the latest canary release. You might also need to restart VSCode for the changes to take effect.
```
Comment (ijjk) [inline errors/middleware-upgrade-guide.md:12]:
```suggestion
If you have ESLint configured, you will need to run `npm i eslint-config-next@canary --save-dev` to upgrade your ESLint configuration to the latest canary release. You might also need to restart VSCode for the changes to take effect.
```
I don't think the force flag needs to be recommended here, did you run into issues without it? We do recommend this be a devDependency though

Comment (steven-tey) [reply errors/middleware-upgrade-guide.md:12]: Yeah I ran into this issue without using `--force`:

CleanShot 2022-06-24 at 22 47 14

Comment (steven-tey) [reply errors/middleware-upgrade-guide.md:12]: VSCode told me to use `--force` and that resolved the issue

Comment (ijjk) [reply errors/middleware-upgrade-guide.md:12]: Hmm I think we can remove the `peerDependency` to resolve that as it's technically not needed

Comment (ijjk) [reply errors/middleware-upgrade-guide.md:12]: Opened PR to fix this here https://github.com/vercel/next.js/pull/38009

Comment (balazsorban44) [reply errors/middleware-upgrade-guide.md:12]: I also don't think this needs to mention `canary`? Maybe say that the version should match the `next` version?

**Comment by balazsorban44** [comment]
See https://github.com/vercel/next.js/pull/38008#discussion_r906661647

File: errors/middleware-upgrade-guide.md [hunk 2]
```diff
@@ -9,6 +9,8 @@ This upgrade guide will help you understand the changes and how to migrate your
 
 You can start upgrading your Middleware usage today with the latest canary release (`npm i next@canary`).
 
+If you have ESLint configured, you will need to run `npm i eslint-config-next@canary --save-dev` to upgrade your ESLint configuration to the latest canary release. You might also need to restart VSCode for the changes to take effect.
```
Comment (ijjk) [inline errors/middleware-upgrade-guide.md:12]:
```suggestion
If you have ESLint configured, you will need to run `npm i eslint-config-next@canary --save-dev` to upgrade your ESLint configuration to ensure the same version is being used as the Next.js version. You might also need to restart VSCode for the changes to take effect.
```

**Review by ijjk — APPROVED** [review]

**Pull request merged**